examples/echo.rs

```diff
@@ -0,0 +1,485 @@
+/// receive a continuous stream of frames from the separate thread. The channel
+/// only has one sender (tx), which is moved to the separate thread. When STDIN
+/// receives an End-of-File (EOF) signal, the sender will be dropped. This
+/// causes the receiver within this actor to become resolved, i.e. completed.
+/// The resolution of the receiver stops this actor.
+///
+/// This actor is implemented as a publisher, in that it does not receive
+/// messages from other actors. Instead, it publishes [`Input`] messages to a
+/// subscriber (recipient).
+///
+/// [`Input`]: struct.Input.html
+struct Stdin<D> {
+    /// The subscriber of the published Input messages.
+    recipient: Recipient<Input>,
+    /// Decodes the stream of bytes from STDIN into frames.
+    decoder: D,
+}
+
+impl<D> Stdin<D>
+    where D: Decoder + Send + Clone + 'static
+{
+    /// Creates a new `Stdin` actor from a decoder and recipient.
+    ///
+    /// The separate thread will be spawn when this actor is started, unlike the
+    /// [`Stdout`] actor, which starts a separate thread when the [`new`] method
+    /// is executed. The separate thread will only stop when it reaches the
+    /// End-of-File (EOF) for STDIN.
+    ///
+    /// Any decoder that decodes data from STDIN into a series of bytes can be
+    /// used. This means any codec provided by the tokio crate under the
+    /// [`tokio::codec`] module can be used, except for the
+    /// [`tokio::codec::LinesCodec`] because it decodes data into strings and
+    /// not bytes. However, the `tokio::codec::LinesCodec` can be used by
+    /// wrapping it in a new type that converts strings into bytes.
+    ///
+    /// [`Stdout`]: struct.Stdout.html
+    /// [`tokio::codec`]: https://tokio-rs.github.io/tokio/tokio/codec/index.html
+    /// [`tokio::codec::LinesCodec`]: https://tokio-rs.github.io/tokio/tokio/codec/struct.LinesCodec.html
+    pub fn new(decoder: D, recipient: Recipient<Input>) -> Self {
+        Stdin {
+            recipient: recipient,
+            decoder: decoder,
+        }
+    }
+}
+
+// A separate thread is started when this actor is started. The separate thread
+// reads bytes from STDIN and sends them to this actor using an internal
+// channel. This implementation of the Actor trait also includes trace
+// logging statements to demonstrate the life-cycle of this actor.
+impl<D> Actor for Stdin<D>
+    where D: Decoder<Item=BytesMut, Error=Error> + Send + Clone + 'static
+{
+    type Context = Context<Self>;
+
+    fn started(&mut self, ctx: &mut Self::Context) {
+        trace!("STDIN started");
+        let (tx, rx) = mpsc::unbounded();
+        let codec = self.decoder.clone();
+        thread::spawn(|| {
+            info!("Begin STDIN thread");
+            // The `io::stdin()` method, which returns an asynchronous Stdin
+            // struct, must be used within a tokio runtime to avoid a
+            // BlockingError.
+            tokio::run(FramedRead::new(io::stdin(), codec)
```

Comment:
Essentially you're making use of three runtimes with, 1 for Stdout, 1 for Stdin and 1 for actix itself.

I don't think it would be good example, instead we should focus on using single runtime of actix.

What you can do is not to use tokio's fs or io with blocking calls, and instead use std's stdin and stdout.
You could similarly read stdin from separate channel and transfer lines using channels